encoder/vaapiencoder_h264.cpp

```diff
@@ -930,6 +930,15 @@ Encode_Status VaapiEncoderH264::reorder(const SurfacePtr& surface, uint64_t time
     bool isIdr = (m_frameIndex == 0 ||m_frameIndex >= keyFramePeriod() || forceKeyFrame);
 
     if (isIdr) {
+        // If the last frame before IDR is B frame, set it to P frame.
```

Comment:
What if last gop do not have enough frame? Suppose you have IBBBBP pattern, but you only have two frame ...
Similar logical should apply to last gop too.